test/action_controller/adapter_selector_test.rb

```diff
@@ -48,6 +53,27 @@ def test_render_skipping_adapter
         get :render_skipping_adapter
         assert_equal '{"attributes":{"name":"Name 1","description":"Description 1","comments":"Comments 1"}}', response.body
       end
+
+      def test_render_using_adapter_selected_by_accept_header_field
+        ActiveModel::Serializer.config.enabled_adapters_by_media_type = true
+        request.headers['Accept'] = "application/vnd.api+json"
+
+        get :render_selecting_adapter_by_accept_header_field
+
+        expected = {
+          data: {
+            id: assigns(:profile).id.to_s,
+            type: "profiles",
+            attributes: {
+              name: "Name 1",
+              description: "Description 1",
+            }
+          }
+        }
+
+        assert_equal expected.to_json, response.body
+      end
```

Comment:
Can you add soem tests covering the override of the media type as well? maybe one using multiple media type declarations too.